Firmware/mmu2.cpp

```diff
@@ -181,6 +182,53 @@ void MMU2::PowerOn(){
     power_on();
 }
 
+enum class MMUErrorPrintStates : uint8_t {
+    NONE           = 0,
+    PAUSE_PRINT    = 1,
+    PARK           = 2,
+    WAITING        = 3,
+    UNPARK         = 4,
+    RESUME_PRINT   = 5,
+};
+
+enum MMUErrorPrintStates ReportErrorPrintState = MMUErrorPrintStates::NONE;
+
+void MMU2::ErrorPrintStateHandler() {
+    switch (ReportErrorPrintState)
+    {
+    case MMUErrorPrintStates::PAUSE_PRINT:
+        start_pause_print = _millis();
+        stop_and_save_print_to_ram(0.0, -default_retraction);
```

Comment:
I still don't follow the technical reason for needing/implementing a different and completely orthogonal pause state here. I don't think you ever want an error condition to behave like a normal "paused" print, that's going to open up a whole complicated mess of potential issues relating to the tune menu, as well as users potentially doing a remote-resume from Octoprint without attending to the actual problem. IMO all MMU errors should follow the same pausing pathway in order to keep the code simpler and reduce future bugs/corner cases.

Reply:
I’m confused how to proceed. I feel like we should at minimum set `isPrintPaused` to `true` etc. So any logic in the `loop()` doesnt do anything unexpected. 

> IMO all MMU errors should follow the same pausing pathway

Do I understand correctly you mean the code shouldn’t care whether we are printing or not?

Reply:
I took out the Resume and Pause states and did quick test. Didn't see any issues. [4cb9f78](https://github.com/DRracer/Prusa-Firmware/pull/48/commits/4cb9f7873698b872278db5c8861ac8378221e327)

Reply:
Yes - in effect all errors are "while printing" or doing something to prepare to print such as a load.

Reply:
Up to now we’ve only been parking while in `manage_response()` which is its own loop and blocks any gcodes from processing during parking. The errors which can happen at any time like the Communication timeout or Protocol error are different in that they happen in the top level `loop()` and we can’t block queued gcodes moves without emptying the planner/command buffer. I’m seeing usually around 6 to 12 commands queued up.